## 75% of end-end tests have passed
```
Total executed tests: 8
Failed tests: 1
Expected to fail tests: 1
Passed tests: 6
```
```
IDs of failed tests: 727229 
```
```
IDs of expected to fail tests: 702843 
```
<h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=seanstrom/keycard-pin-error-message&PR_ID=21691&APK_NAME=21691.apk&TR_CASE_IDS=727229">Rerun failed tests</a></li><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3735542">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Button` by `accessibility id`: `network-dropdown` </div><div>Device 2: Tap on found: Button </div></blockquote></p>

```
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:703: in urlopen
    httplib_response = self._make_request(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:449: in _make_request
    six.raise_from(e, None)
<string>:3: in raise_from
    ???
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:444: in _make_request
    httplib_response = conn.getresponse()
/usr/lib/python3.10/http/client.py:1375: in getresponse
    response.begin()
/usr/lib/python3.10/http/client.py:318: in begin
    version, status, reason = self._read_status()
/usr/lib/python3.10/http/client.py:287: in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
E   http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/requests/adapters.py:486: in send
    resp = conn.urlopen(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:787: in urlopen
    retries = retries.increment(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/util/retry.py:550: in increment
    raise six.reraise(type(error), error, _stacktrace)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/packages/six.py:769: in reraise
    raise value.with_traceback(tb)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:703: in urlopen
    httplib_response = self._make_request(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:449: in _make_request
    six.raise_from(e, None)
<string>:3: in raise_from
    ???
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:444: in _make_request
    httplib_response = conn.getresponse()
/usr/lib/python3.10/http/client.py:1375: in getresponse
    response.begin()
/usr/lib/python3.10/http/client.py:318: in begin
    version, status, reason = self._read_status()
/usr/lib/python3.10/http/client.py:287: in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
 ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

During handling of the above exception, another exception occurred:
critical/test_wallet.py:130: in test_wallet_send_eth
    sender_balance, receiver_balance, eth_amount_sender, eth_amount_receiver = self._get_balances_before_tx()
critical/test_wallet.py:39: in _get_balances_before_tx
    sender_balance = self.network_api.get_balance(self.sender['wallet_address'])
../support/api/network_api.py:52: in get_balance
    balance = self.send_etherscan_request(params)
../support/api/network_api.py:29: in send_etherscan_request
    response = requests.get(url=self.network_url, params=params).json()
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/requests/api.py:73: in get
    return request("get", url, params=params, **kwargs)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/requests/api.py:59: in request
    return session.request(method=method, url=url, **kwargs)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/requests/sessions.py:589: in request
    resp = self.send(prep, **send_kwargs)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/requests/sessions.py:725: in send
    history = [resp for resp in gen]
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/requests/sessions.py:725: in <listcomp>
    history = [resp for resp in gen]
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/requests/sessions.py:266: in resolve_redirects
    resp = self.send(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/requests/sessions.py:703: in send
    r = adapter.send(request, **kwargs)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/requests/adapters.py:501: in send
    raise ConnectionError(err, request=request)
E   requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
```

<br/><br/></td></tr></tbody></table></details><h3>Expected to fail tests (1)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3735540">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Skipped due to waku issue on staging fleet]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (6)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3735543">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3735538">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/441954c55acf472d895436c77fa3b486?auth=2d453b0ad7b9fc01449ab836a7fb7434#47">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3735541">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/441954c55acf472d895436c77fa3b486?auth=2d453b0ad7b9fc01449ab836a7fb7434#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3735544">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3735545">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3735539">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/475ffba3f1d249338adff6a3268486db?auth=b9ab798a71bcc628f0032778c38be614#107">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2da93a27b49642bdbf164fa0df58eb48?auth=cd27697917f6259fd1383462feced55a#94">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/keycard/events.cljs
+++ b/src/status_im/contexts/keycard/events.cljs
@@ -45,6 +45,7 @@
      (when-not (or tag-was-lost? (nil? pin-retries-count))
        {:db (-> db
                 (assoc-in [:keycard :application-info :pin-retry-counter] pin-retries-count)
+                (assoc-in [:keycard :pin :text] "")
                 (assoc-in [:keycard :pin :status] :error))
         :fx [[:dispatch [:keycard/disconnect]]
              (when (zero? pin-retries-count)

--- a/src/status_im/contexts/keycard/pin/enter/view.cljs
+++ b/src/status_im/contexts/keycard/pin/enter/view.cljs
@@ -14,5 +14,4 @@
       {:icon-name :i/close
        :on-press  events-helper/navigate-back}]
      [quo/page-top {:title (i18n/label :t/enter-keycard-pin)}]
-     [rn/view {:style {:flex 1}}]
      [keycard.pin/auth {:on-complete on-complete}]]))

--- a/src/status_im/contexts/keycard/pin/view.cljs
+++ b/src/status_im/contexts/keycard/pin/view.cljs
@@ -29,7 +29,7 @@
         :info                  (when error?
                                  (if (not (string/blank? error-message))
                                    error-message
-                                   (i18n/label :t/pin-retries-left {:number pin-retry-counter})))}]
+                                   (i18n/label-pluralize pin-retry-counter :t/pin-retries-left)))}]
       [quo/numbered-keyboard
        {:delete-key? true
         :on-delete   #(rf/dispatch [:keycard.pin/delete-pressed])

--- a/src/status_im/contexts/keycard/pin/view.cljs
+++ b/src/status_im/contexts/keycard/pin/view.cljs
@@ -13,8 +13,14 @@
         pin-retry-counter              (rf/sub [:keycard/pin-retry-counter])
         error?                         (or error? (= status :error))]
     (rn/use-unmount #(rf/dispatch [:keycard.pin/clear]))
-    [rn/view {:padding-bottom 12 :flex 1}
-     [rn/view {:flex 1 :justify-content :center :align-items :center :padding 34}
+    [rn/view
+     {:style {:flex           1
+              :flex-direction :row
+              :align-items    :flex-end}}
+     [rn/view
+      {:style {:flex           1
+               :gap            34
+               :padding-bottom 12}}
       [quo/pin-input
        {:blur?                 false
         :number-of-pins        constants/pincode-length
@@ -23,9 +29,9 @@
         :info                  (when error?
                                  (if (not (string/blank? error-message))
                                    error-message
-                                   (i18n/label :t/pin-retries-left {:number pin-retry-counter})))}]]
-     [quo/numbered-keyboard
-      {:delete-key? true
-       :on-delete   #(rf/dispatch [:keycard.pin/delete-pressed])
-       :on-press    #(rf/dispatch [:keycard.pin/number-pressed % constants/pincode-length
-                                   on-complete])}]]))
+                                   (i18n/label :t/pin-retries-left {:number pin-retry-counter})))}]
+      [quo/numbered-keyboard
+       {:delete-key? true
+        :on-delete   #(rf/dispatch [:keycard.pin/delete-pressed])
+        :on-press    #(rf/dispatch [:keycard.pin/number-pressed % constants/pincode-length
+                                    on-complete])}]]]))

--- a/translations/en.json
+++ b/translations/en.json
@@ -1981,7 +1981,10 @@
   "pin-one-attempt-blocked-before": "Be careful, you have only",
   "pin-one-attempt-frozen-after": "before your Keycard gets frozen",
   "pin-one-attempt-frozen-before": "Be careful, you have only",
-  "pin-retries-left": "{{number}} attempts left",
+  "pin-retries-left": {
+    "one": "Wrong PIN, 1 attempt left",
+    "other": "Wrong PIN, {{count}} attempts left"
+  },
   "pin-to-channel": "Pin to the channel",
   "pin-to-chat": "Pin to the chat",
   "Pinned-a-message": "Pinned a message",

